File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {


        if (s.length() == 0) {
            return false;
        }

        Stack<Character> stack = new Stack<>();

        HashSet<Character> hashSet = new HashSet<>();
        hashSet.add('[');
        hashSet.add('(');
        hashSet.add('{');

        for (int i = 0; i < s.length(); i++) {
            Character character = s.charAt(i);

            if (hashSet.contains(character)) {
                stack.add(character);
            }

            else {

                if (stack.isEmpty()) {
                    return false;
                }

                Character lastElement = stack.lastElement();
                if ((lastElement.equals('[') && character.equals(']')) || (lastElement.equals('(') && character.equals(')')) || (lastElement.equals('{') && character.equals('}'))) {
                    stack.pop();
                }
                else {
                    return false;
                }
            }
        }

        return stack.isEmpty();




 
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.js
var isValid = function (s) {
  if (s === "") {
    return false;
  }

  const leftBrackets = new Set(["[", "{", "("]);
  const stack = [];
    
  for (let i = 0; i < s.length; i++) {
    const character = s[i];

    if (leftBrackets.has(character)) {
      stack.push(character);
    } else {
      if (stack.length === 0) {
        return false;
      }

      const lastElement = stack[stack.length - 1];
      if ((lastElement === "[" && character === "]") || (lastElement === "(" && character === ")") || (lastElement === "{" && character === "}")
      ) {
        stack.pop();
      } else {
        return false;
      }
    }
  }
  return stack.length === 0;
};
